mirrord/layer/src/connection.rs

```diff
@@ -1,130 +1,91 @@
-///
-/// Used in [`connect`] (same as [`handle_error`]), but only when `LayerConfig::operator` is
-/// set to `true` and an operator is installed in the cluster.
-fn handle_operator_error(err: OperatorApiError) -> ! {
-    graceful_exit!("{FAIL_CREATE_AGENT}{FAIL_STILL_STUCK} with error {err}")
-}
-
-/// Initializes the agent pod with [`KubernetesAPI`], and connects to it, returning an API that the
+/// Connects to the internal proxy in given `SocketAddr`
 /// layer uses to communicate with it, in the form of a [`Sender`] for [`ClientMessage`]s, and a
 /// [`Receiver`] for [`DaemonMessage`]s.
-///
-/// ## Direct connection
-///
-/// There is support for connecting directly to an existing pod (instead of going through
-/// `KubernetesAPI::create`) by setting `LayerConfig::connect_tcp`.
-///
-/// This is used for mirrord tests only.
-///
-/// ## Operator
-///
-/// The [`OperatorApi`] takes over the connection procedure if `LayerConfig::operator` is
-/// set to `true` and an operator is installed in the cluster. We use `OperatorApi::discover`
-/// instead of [`KubernetesAPI`].
-pub(crate) async fn connect(
-    config: &LayerConfig,
-) -> (Sender<ClientMessage>, Receiver<DaemonMessage>) {
-    let progress = NoProgress;
-
-    if let Some(address) = &config.connect_tcp {
-        Connection(address)
-            .connect(&progress)
-            .await
-            .unwrap_or_else(|err| handle_error(err, config))
-    } else if config.operator && let Some(connection) = OperatorApi::discover(config).await.transpose() {
-        connection.unwrap_or_else(|err| handle_operator_error(err))
-    } else {
-        let k8s_api = KubernetesAPI::create(config)
-            .await
-            .unwrap_or_else(|err| handle_error(err, config));
-
-        let agent_ref = {
-            if let (Some(pod_agent_name), Some(agent_port)) =
-                (&config.connect_agent_name, config.connect_agent_port)
-            {
-                info!(
-                    "Reusing existing agent {:?}, port {:?}",
-                    pod_agent_name, agent_port
-                );
-                (pod_agent_name.to_owned(), agent_port)
-            } else {
-                let (pod_agent_name, agent_port) = tokio::time::timeout(
-                    Duration::from_secs(config.agent.startup_timeout),
-                    k8s_api.create_agent(&progress),
-                )
-                .await
-                .unwrap_or(Err(KubeApiError::AgentReadyTimeout))
-                .unwrap_or_else(|err| handle_error(err, config));
-                info!("Created new agent {pod_agent_name}, port {agent_port}");
-
-                // Set env var for children to re-use.
-                std::env::set_var("MIRRORD_CONNECT_AGENT", &pod_agent_name);
-                std::env::set_var("MIRRORD_CONNECT_PORT", agent_port.to_string());
+pub(crate) async fn connect(addr: SocketAddr) -> (Sender<ClientMessage>, Receiver<DaemonMessage>) {
```

Comment:
I would suggest changing this to`connect_to_proxy`, as for 1 second there I thought this was the `connect` hook :sweat_smile: